src/Core/Grid/Definition/Factory/DeleteActionTrait.php

```diff
@@ -41,25 +41,33 @@ protected function buildDeleteAction(
         string $deleteRouteParamName,
         string $deleteRouteParamField,
         string $method = 'POST',
-        array $extraRouteParams = []
+        array $extraRouteParams = [],
+        array $options = [],
+        array $modalOptions = []
     ): RowActionInterface {
         return (new SubmitRowAction('delete'))
             ->setName($this->trans('Delete', [], 'Admin.Actions'))
             ->setIcon('delete')
-            ->setOptions([
-                'route' => $deleteRouteName,
-                'route_param_name' => $deleteRouteParamName,
-                'route_param_field' => $deleteRouteParamField,
-                'extra_route_params' => $extraRouteParams,
-                'confirm_message' => $this->trans('Are you sure you want to delete the selected item(s)?', [], 'Admin.Global'),
-                'method' => $method,
-                'modal_options' => new ModalOptions([
-                    'title' => $this->trans('Delete selection', [], 'Admin.Actions'),
-                    'confirm_button_label' => $this->trans('Delete', [], 'Admin.Actions'),
-                    'close_button_label' => $this->trans('Cancel', [], 'Admin.Actions'),
-                    'confirm_button_class' => 'btn-danger',
-                ]),
-            ])
+            ->setOptions(array_merge(
```

Comment:
Have you thought about using native PHP function `array_merge_recursive` ?

Reply:
Nope. `array_merge_recursive` will not work in our case.

Example : http://sandbox.onlinephpfunctions.com/code/8ba947a455a78865fbf2ecfa163982e2b6cbf8ce

If the key exists, that will create an array.

Reply:
very true !